docker-compose.yml

```diff
@@ -104,12 +104,25 @@ services:
       retries: 20
       start_period: 0s
     networks: ['agent-net']
-
+  postgres-veritable:
+    container_name: ${VERITABLE_PG_HOST}
+    image: postgres:16.2-alpine
+    restart: on-failure
+    ports:
+      - ${VERITABLE_PG_PORT}:5432
```

Comment:
This exposes the port on the host, which we shouldn't have to do (unless there's a testing use case, in which case this should be defined only in `docker-compose.test.override.yml`). In fact, with the current configuration, this would lead to a host port conflict, as all agents try to expose the same port on the host.

Reply:
so this means that we would specify no ports?

Reply:
I take this comment back - we need these ports to be exposed purely for development mode (described above). And although we can set a single default `VERITABLE_PG_HOST` for all agents when using `agent-net` network, we need different default `VERITABLE_PG_PORT` values for each to avoid host port conflicts (see existing `.env*` files for examples. In fact, `VERITABLE_PG_PORT` is better split across two env vars, `VERITABLE_PG_PORT` to be used by the `portal` service, and `VERITABLE_PG_PORT_HOST` to define which port should be exposed on the host.

Essentially, we have 2 scenarios:
1. nice-agent-portal run normally (using docker compose, on the `nice-net` and `agent-net` networks). The compose definition should load these env vars (using `.env` for shared default values and `.env.nice-agent-<role>` to override agent-specific values):
    - On `portal` service:
        - VERITABLE_PG_HOST: "postgres-veritable"
        - VERITABLE_PG_PORT: 5432
    - On `postgres-veritable` service:
        - VERITABLE_PG_PORT_HOST: unique port for each role, follow existing port numbering schema, so it would be 3030, 3031 and 3032 for alice, bob and issuer respectively
2. nice-agent-portal run on host (using `npm`, no access to the docker networks so we need to interface through the host network), all other services run through docker compose. The compose definition should load these env vars (overridden by defining in `.env.nice-agent-<role>.dev`):
    - On `portal` service:
        - VERITABLE_PG_HOST: "localhost"
        - VERITABLE_PG_PORT: should be the same as VERITABLE_PG_PORT_HOST